SQL 26th sep/26sep'24.txt

```diff
@@ -0,0 +1,78 @@
+1.) 
+create database employee2;
+use employee2;
+CREATE TABLE Employees (  EmployeeID INT PRIMARY KEY,FirstName VARCHAR(50), LastName VARCHAR(50),  Department VARCHAR(50),Salary DECIMAL(10, 2),HireDate DATE);
+select * from Employees;
+insert into Employees values (1,'aaa','Abin','Marketing',15000, '2002-07-20');
+insert into Employees values (2,'bbb','beni','sales ', 60000, '2020-08-01');
+insert into Employees values (3,'ccc','Cyril','IT',87000, '2019');
+insert into Employees values (4,'John','daniel','marketing ', 60000, '2019-10-09');
+insert into Employees values (5,'John','Saniel','marketing ', 75000, '2007-10-09');
+insert into Employees values (6,'Krish','Saniel','IT', 50000, '2010-10-09');
+
+create table students(name nvarchar,rollno int PRIMARY KEY,name nvarchar(30));
+insert into students values (1,N'അഞ്ജലി');
+--Retrieve all employees who work in Sales, Marketing, or IT departments.
+select * from employees where Department in ('cse','marketing','sales');
+
+--Find all employees with salaries ranging from $50,000 to $75,000 (inclusive).
+select * from employees where salary between 50000 and 75000;
+
+--List all employees whose last name begins with the letter 'S'.
+select * from employees where LastName like 's%';  
+
+--Display all employees with exactly five letters in their first name.
+select * from employees where FirstName like '_____';
+
+--Find employees whose last name starts with either 'B', 'R', or 'S'.
+select * from employees where lastname like 'r%' or lastname like 's%'or lastname like 'b%';
+
+--Retrieve all employees whose first name begins with any letter from 'A' through 'M'.
+select * from employees where FirstName like '[A-M]%';
+
+--List employees whose last name doesn't start with a vowel (A, E, I, O, U).
+select * from employees where lastname not LIKE 'a%' and  lastname not LIKE 'e%' and  lastname not LIKE  'i%' and lastname not LIKE  'u%' and  lastname not LIKE 'o%';
+
+--Identify employees earning more than $80,000 annually. 
+select * from employees where salary>80000;
+
+--Find employees who joined the company before 2020.
+select * from employees where HireDate <='2020-01-01'; 
+ 
+--List all employees not named 'John' (first name).
+select * from employees where firstname not like 'John';
+
+--Identify Marketing department employees earning $60,000 or less who were hired after June 30, 2019.
+select * from employees where department='marketing' and salary<=60000 and HireDate >='2013-06-30';
```

Comment:
this should be "Or" condition right ?

Reply:
as of what i got from the question it is to print marketing employees whose salary was either equal or less than 60,000 and also they should be hired after 30th june 2013.
'or' will give only the one who follows either of the condition , in question its asking about both the conditions so and works .